src/fillTank.test.js

```diff
@@ -1,11 +1,147 @@
 'use strict';
 
 describe('fillTank', () => {
-  // const { fillTank } = require('./fillTank');
+  const { fillTank } = require('./fillTank');
 
-  it('should ', () => {
+  it('should ', () => {});
 
+  // write tests here
+
+  it(`should be declared`, () => {
+    expect(fillTank).toBeInstanceOf(Function);
   });
 
-  // write tests here
+  it(`should return nothing`, () => {
+    const customer = {
+      money: 3000,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 8,
+      },
+    };
+
+    expect(fillTank(customer)).toBeUndefined();
+  });
+
+  it(`should ordered full tank if amount is not given`, () => {
+    const customer = {
+      money: 3000,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 10,
+      },
+    };
+
+    fillTank(customer, 5);
+
+    expect(customer).toEqual({
+      money: 2850,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 40,
+      },
+    });
+  });
+
+  it(`should not pour more than tank can accommodate`, () => {
+    const customer = {
+      money: 3000,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 10,
+      },
+    };
+
+    fillTank(customer, 5, 50);
+
+    expect(customer).toEqual({
+      money: 2850,
+      vehicle: {
+        maxTankCapacity: 40,
+        fuelRemains: 40,
+      },
+    });
+  });
```

Comment:
In the test it should not pour more than tank can accommodate, you are passing a third parameter, 50, as the amount to be poured. However, the third parameter isn't explained in the test description or logic.